cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/writer/MetadataMessage.java

```diff
@@ -37,41 +34,25 @@
     LINEAGE,
     USAGE,
     WORKFLOW_TOKEN,
-    WORKFLOW_STATE
+    WORKFLOW_STATE,
+    PROFILE_UPDATE,
+    PROILE_REMOVE
   }
 
   private final Type type;
-  private final ProgramId programId;
-  @Nullable
-  private final String runId;
+  private final EntityId entityId;
```

Comment:
this is a backwards incompatible change, which means any unprocessed messages on the topic prior to 5.0 will cause failures. Or if the offset for the topic is deleted, old messages will be re-processed and cause failures. 

To remove these issues, you would have to make programId nullable, and just record all your required information in the payload.

Reply:
this class and the MetadataSubscriberService are actually added in 5.0, so is it fine to make direct change here?

Reply:
oh if these are all new then it's ok to change it.